File: C++/50/50.cpp
#include <iostream>
using namespace std;
class Solution
{
public:
    double myPow(double x, int n)
    {
        double result = 1;
        bool flag = n > 0;
        long int pow = n;
        if (n < 0)
            pow = -pow;
        while (pow > 0)
        {
            if (pow % 2 == 0)
            {
                pow /= 2;
                x *= x;
            }
            else
            {
                pow = --pow / 2;
                result *= x;
                x *= x;
            }
        }
        return flag ? result : 1 / result;
    }
};
int main()
{
    Solution so;
    cout << so.myPow(2, 4) << endl;
    system("pause");
    return 0;
}

File: Python/50/50.py
class Solution(object):
    def myPow(self, x, n):
        """
        :type x: float
        :type n: int
        :rtype: float
        """
        flag = 1
        result = 1
        if n < 0:
            flag = 0
            n = -n
        while n > 0:
            if n % 2 == 0:
                x = x*x
                n = n//2
            else:
                n = n-1
                result = x*result
                n = n//2
                x = x*x
        if flag == 1:
            return result
        else:
            return 1/result



